### PR TITLE
Protect: Fix last_checked conversions

### PR DESCRIPTION
**Pull request by dkmyta**
Fixes #
The `last_checked` value we retrieve from the scan API is a reformatted UTC date time value - in the scanning UI its treated as though it is in local time and therefor double converted based on the WP installation timezone settings.

## Proposed changes:

* Convert the timestamp to local time before passing it to the WP timezone conversion function

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1439

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch running Jurassic Tube with Protect installed and a scan upgrade
* Set your WP installation timezone settings to your local time
* Within the `Summary` component log out the value of `dateI18n( 'F jS H:i:s', lastChecked )`
   * Include `H:i:s` for better clarity
* Load the scan UI after running a successful scan
* Verify that the date value provided is incorrect
* Log out `dateI18n( 'F jS H:i:s', lastCheckedLocalTimestamp )` and verify that it aligns with the time of the last successful in the timezone you have set
* Update your installation timezone settings and ensure the value changes and the UI updates accordingly

### CODE REVIEW

**Comment by nateweller** [comment]
I'm seeing the following code in `class-scan-status.php:154-159`:

```
if ( ! empty( $scan_data->most_recent->timestamp ) ) {
	$date = new \DateTime( $scan_data->most_recent->timestamp );
	if ( $date ) {
		$status->last_checked = $date->format( 'Y-m-d H:i:s' );
	}
}
```

Due to the server processing with `DateTime`, my expectation would be that the timestamp from the server would be in the timezone set by site settings. Testing on a site, the `last_checked` value does change as I change the site timezone setting.

If the objective is to provide the timestamp from the server as UTC, and display it on the front end in the timezone of the current browser, I believe we will need to update the server side as well, i.e. `$date->setTimezone( new \DateTimeZone( 'UTC' ) )` prior to calling `$date->format()`.

File: projects/plugins/protect/src/js/components/summary/index.jsx [hunk 1]
```diff
@@ -38,7 +41,7 @@ const Summary = () => {
 						{ sprintf(
 							/* translators: %s: Latest check date  */
 							__( 'Latest results as of %s', 'jetpack-protect' ),
-							dateI18n( 'F jS', lastChecked )
+							dateI18n( 'F jS', lastCheckedLocalTimestamp )
```
Comment (nateweller) [inline projects/plugins/protect/src/js/components/summary/index.jsx:44]: The [dateI18n](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/#datei18n) function actually accepts a third argument to specify the timezone or UTC offset :+1:

**Review by nateweller — CHANGES_REQUESTED** [review]
See above comments

**Comment by dkmyta** [comment]
>Due to the server processing with DateTime, my expectation would be that the timestamp from the server would be in the timezone set by site settings. Testing on a site, the last_checked value does change as I change the site timezone setting.

Surfacing what we discussed privately for posterity. It seems that the timestamp we receive from the Scan API `$scan_data->most_recent->timestamp` always specifies the timezone as UTC `+00:00` so the DateTime function respects that in the conversion. In my testing, I did not recognize a change to this value when updating the WordPress installation timezone settings.

File: projects/plugins/protect/src/js/components/summary/index.jsx (hunk 1 above)
Comment (dkmyta) [reply projects/plugins/protect/src/js/components/summary/index.jsx:44]: If we supply this third argument, it appears the internal mechanism that converts it to the timezone specified in the WordPress installation settings is not used.

**Review by nateweller — APPROVED** [review]

**Pull request merged**